frontend/src/components/MonacoEditor/sqlParser.ts

```diff
@@ -0,0 +1,24 @@
+import { Parser, AST } from "node-sql-parser";
+
+type ParseResult = {
+  data: AST | AST[] | null;
+  error: Error | null;
+};
+
+const parseSQL = (sql: string): ParseResult => {
+  const parser = new Parser();
+
+  try {
+    const ast = parser.astify(sql);
+    return { data: ast, error: null };
+  } catch (error) {
+    return { data: null, error: error as Error };
+  }
+};
+
+export const isSelectStatement = (sql: string) => {
+  const { data } = parseSQL(sql);
+  return data?.type.toLowerCase() === "select";
```

Comment:
I'm curious about the `type` that given by node-sql-parser is still 'select' if I use a subquery in SELECT.
```sql
SELECT * FROM test WHERE id in (INSERT INTO test ...)
```